manager_test.py

```diff
@@ -1,82 +1,79 @@
 #!/usr/bin/env python3
 
+# TODO: migrate away from unittest patch
 from unittest.mock import patch
-import unittest
+
+import pytest
 import manager
 import job
 
-class TestManager(unittest.TestCase):
-    def setUp(self):
-        self.sched_cfg = {
-                'tmpdir_stagger_phase_major': 3,
-                'tmpdir_stagger_phase_minor': 0,
-                'tmpdir_max_jobs': 3 }
-        self.dir_cfg = {
-                'tmp_overrides': {
-                        '/mnt/tmp/04': {
-                                'tmpdir_max_jobs': 4 }}}
+@pytest.fixture
+def sched_cfg():
```

Comment:
This is perfectly correct.  But maybe you go and write a test like this.

```python
def test_stuff():
    assert len(sched_cfg) == 3
```

And oops, you are accessing the fixture function instead of having pytest pass you the fixture value because you forgot to put `sched_cfg` into your test function's parameter list.

You can avoid this by having your fixtures in another file.  Sometimes they are common and this makes sense.  I also end up doing this.

```python
@pytest.fixture(name='sched_cfg')
def sched_cfg_fixture():
```

Not particularly strongly suggesting a change here.  Just exposing a couple aspects of fixtures.

Reply:
Thanks.  My brain is pretty full, but I'll try to register this for the future.

Reply:
What's the second code example demonstrating?  I don't think I understand.

Reply:
The name of the fixture function in the second code block is `sched_cfg_fixture` so when you make the error in the first code block you'll get a straight `NameError` as you might 'normally expect' rather than `TypeError: object of type 'function' has no len()`.

Reply:
Oh, the second is just a naming convention?

Reply:
Along with the ability to set the fixture name via a parameter to the decorator rather than only by the name of the function itself.

Reply:
OK.  I will have to read more about this one.